### PR TITLE
Add benchmarking scaffolding plus a benchmark test for Program

### PR DESCRIPTION
**Pull request by ibgreen**
Add benchmarking support

Initial test case confirms that we need to get shader caching sorted asap.

Even for trivial shaders we see a big cost for recompiling shaders. For many instances of a big shader (like multiple copies of a deck.gl layer), the cost will be much higher.

```
Starting bench...
* Program from shader source x 121 ops/sec ±1.64% (52 runs sampled)
* Program from cached shaders x 180 ops/sec ±1.95% (58 runs sampled)
```

### CODE REVIEW

File: src/packages/math/src/euler.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import MathArray from './math-array';
 import {checkNumber, clamp} from './common';
-import {Matrix4} from './matrix4';
-import {Quaternion} from './quaternion';
-import {Vector3} from './vector3';
+import Matrix4 from './matrix4';
+import Quaternion from './quaternion';
+import Vector3 from './vector3';
```
Comment (Pessimistress) [inline src/packages/math/src/euler.js]: +1 for fixing this class, has been broken for a while.
But should be on a separate PR?

File: demo/index.html [hunk 2]
```diff
@@ -0,0 +1,306 @@
+<!DOCTYPE html>
```
Comment (Pessimistress) [inline demo/index.html]: What's changed with this file, just moving?

File: src/packages/math/src/euler.js (hunk 1 above)
Comment (ibgreen) [inline src/packages/math/src/euler.js]: > But should be on a separate PR?

Yes, I suppose that could have made sense. This PR could have been broken into three really small pieces. I'll be a little less trigger happy with the merge button going forward.

File: demo/index.html (hunk 2 above)
Comment (ibgreen) [inline demo/index.html]: Yes webpack-dev-server needs a short index.html in the root directory to run the bench tests.

Since examples are broken on master and we are moving to a demo structure anyway, I moved this index file into the intended target directory now.
